acme/acme/messages.py

```diff
@@ -258,6 +259,27 @@ class ResourceBody(jose.JSONObjectWithFields):
     """ACME Resource Body."""
 
 
+class ExternalAccountBinding:
+    """ACME External Account Binding"""
+
+    @classmethod
+    def from_data(cls, key=None, kid=None, hmac=None):
+        """Create External Account Binding Resource from contact details, kid and hmac."""
+
+        key_json = json.dumps(key.to_partial_json())
```

Comment:
Not sure I understand here. Doesn't the payload argument to JWS.sign() have to be a string?

Reply:
The conversion to a string is handled automagically by different classes in josepy. The same `json.dumps(x.to_partial_json())` is being done behind the scenes.

Thanks to @bmw for doing the digging here!

Reply:
Sorry. I wasn't very clear in my explanation to @ohemorange here. I think this line without changes is correct.

For context, I was initially unsure because we pass this same value in the `jwk` header all the time and don't have any code like this in acme/certbot, but that's because we have a special [Header class in josepy](https://github.com/certbot/josepy/blob/v1.1.0/src/josepy/jws.py#L38) that handles this for us. Passing this value in the `payload` is a one time thing and ACME specific rather than being generic to JOSE, so I think this approach works just fine.

Reply:
Ok, thanks guys! Resolving this one now then.